Realm/RLMArrayLinkView.mm

```diff
@@ -90,7 +90,7 @@ static void throwError() {
         throw;
     }
     catch (realm::InvalidTransactionException const&) {
-        @throw RLMException(@"Cannot modify persisted RLMArray outside of a write transaction");
+        @throw RLMException(@"Cannot modify managed RLMArray outside of a write transaction");
```

Comment:
Like most of Foundation, we don't officially support doing anything meaningful after catching the exceptions we throw at all, much less doing things based on the message.